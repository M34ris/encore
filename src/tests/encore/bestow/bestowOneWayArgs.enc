local class IntBox
   var bar : int

   def init(bar : int) : unit
      this.setInt(bar)
   end

   def setInt(bar : int) : unit
      this.bar = bar
      println(this.bar)
   end
end

active class Foo
   var box : IntBox

   def init(bar : int) : unit
      this.box = new IntBox(bar)
   end

   def bestowBox() : Bestow[IntBox]
      bestow(this.box)
   end
end

active class Main
   def main() : unit
      var b = new Foo(7)
      var box = get(b ! bestowBox())
      box ! setInt(11)
      box ! setInt(42)
   end
end
