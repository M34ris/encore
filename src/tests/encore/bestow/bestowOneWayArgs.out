7
11
42